### DISCUSSION TITLE
Disable Email verification OR assume email is preverified OR email is optional..

### DISCUSSION THREAD

**Discussion by zetoqu** [zitadel/zitadel#4145] [Q&A] [answered]
Hi,
    during self registration, i would like to make email as optional. How can i do that ?
   my workflow assumes that user doesn't has emails.
   one of the following will get my job done but i can't find a way to do any of these.
    1) Make email optional.
    2) Disable email verification completely.
    3) or assume all emails are preverfied .

Thank you

**Comment by hifabienne** [comment, marked as answer]
Hi @zetoqu 

At the moment emails can't be marked as optional they are always required.
Disabling the email verification would not solve your problem, because ZITADEL requires verified email addresses.
When using the API or the management Console to create users it is possible to set the email addresses as verified.

**Reply by zetoqu** [reply]
Hi, 
      Disabling email verification will definitely solve my use-case, where users are not supposed to require email address. Can you please help me disabe email verification ( or automatic verification at creation) and thus zitadel  will assume that all email address are automatically verified without actual verification.  This will be required so as self-registration users can register instantly.
Thank you for your help.

**Reply by hifabienne** [reply]
Hi,
Sorry maybe I didn't write clearly. At the moment this is not possible in ZITADEL.
You could configure an SMTP that is not correct, so no E-Mails are sent. But because ZITADEL always requires verified emails your users would not be able to login after (because they are not able to verify email adresses).

At the moment the solution for your problem would be to add users via management api and set the email verified flag to true.

**Comment by Lan-Phan** [comment]
Hello @hifabienne ,

Is there any update to have email optional in Zitadel? AFAIK, there are so many applications use phone number as a username and verify its via SMS or their Instance Message App (Line, WhatsApp ...), they don't need email at all.

**Reply by hifabienne** [reply]
Hei @Lan-Phan 
Not yet. We do have an issue for removing required fields on the user, and there are other issues linked in there, that have to be implemented first. It is still planned to implement, but we had some higher prioritise at the moment.
You can subscribe to this issue to stay up to date with when it is implemented: https://github.com/zitadel/zitadel/issues/4386

**Comment by ben-epp** [comment]
Hey @zetoqu ,

I don't now if it is still relevant for you, but maybe for others searching for a solution to similar problems.
I have also searched for the option to set all emails from our IDP as verified, because we are using Windows AD locally and there is no such thing as EmailVerified in AD.

So it would be the same as your option 3).

I found the solution in ZITADELs docs:
This can be done in zitadel with [ZITADEL action](https://zitadel.com/docs/apis/actions/introduction) .
For our setup it is the perfect solution.
Now I am able to verify the email and phone for our users with an action like in the[ example](https://zitadel.com/docs/guides/integrate/identity-providers/additional-information) with the Trigger "Post Authentication".

In your case you could set a dummy email and verify it with the script. 
Also check the "Disable Email Login" Checkbox in the menu "Login Behavior and Security" of your organisation.
